### ISSUE TITLE
Add regularization to GP hyperparam optimization

### ISSUE THREAD

**Issue by somu15** [idaholab/moose#23026]
## Reason
Regularization by including prior distributions over the hyperparams of GP in the loss function aids in the optimization

## Design
The log of hyperparam priors are all fixed to a Normal distribution centered at zero. An optional standard deviation value is requested. By default, this value is set to 1000.0 and this shouldn't have any impact on the existing behavior and results. It's when the standard deviation value is lowered the effect of regularization kicks in.

## Impact
No anticipated impacts to existing objects.
